src/parsers/manifest/dash/node_parsers/index.ts

```diff
@@ -215,6 +224,55 @@ const getLastLiveTimeReference = (
   return representationsMin;
 };
 
+/**
+ * From parsed media url, build a representation based media url.
+ * @param {Object} segmentIndex
+ * @param {string} repURL
+ * @param {string} repId
+ * @param {number} repBitrate
+ */
+function patchRepresentationMediaURL(
+  segmentIndex: IParsedSegmentBase,
+  repURL: string,
+  repId: string,
+  repBitrate: number
+): IParsedSegmentBase;
+function patchRepresentationMediaURL(
+  segmentIndex: IParsedSegmentList,
+  repURL: string,
+  repId: string,
+  repBitrate: number
+): IParsedSegmentList;
+function patchRepresentationMediaURL(
+  segmentIndex: IParsedSegmentTemplate|IParsedSegmentTimeline,
+  repURL: string,
+  repId: string,
+  repBitrate: number
+): IParsedSegmentTemplate|IParsedSegmentTimeline;
+function patchRepresentationMediaURL(
+  segmentIndex:
+    IParsedSegmentBase|IParsedSegmentList|IParsedSegmentTemplate|IParsedSegmentTimeline,
+  repURL: string,
+  repId: string,
+  repBitrate: number
+): IParsedSegmentBase|IParsedSegmentList|IParsedSegmentTemplate|IParsedSegmentTimeline {
+  const newIndex = JSON.parse(JSON.stringify(segmentIndex));
+  newIndex.media = replaceRepresentationDASHTokens(
+    repURL + (segmentIndex.media || ""),
```

Comment:
Are we sure that we should do a concatenation here and not ``resolveURL``?

Can ``media`` be a full path or is it always just a filename (with no leading slash or point-slash)